test/e2e/framework/service/jig.go

```diff
@@ -558,20 +616,14 @@ func (j *TestJig) WaitForNewIngressIP(ctx context.Context, existingIP string, ti
 func (j *TestJig) ChangeServiceNodePort(ctx context.Context, initial int) (*v1.Service, error) {
 	var err error
 	var service *v1.Service
-	for i := 1; i < NodePortRange.Size; i++ {
-		offs1 := initial - NodePortRange.Base
-		offs2 := (offs1 + i) % NodePortRange.Size
-		newPort := NodePortRange.Base + offs2
-		service, err = j.UpdateService(ctx, func(s *v1.Service) {
-			s.Spec.Ports[0].NodePort = int32(newPort)
-		})
-		if err != nil && strings.Contains(err.Error(), errAllocated.Error()) {
-			framework.Logf("tried nodePort %d, but it is in use, will try another", newPort)
-			continue
-		}
-		// Otherwise err was nil or err was a real error
-		break
+	newport, ok := j.GetUnusedStaticNodePortAndReserve()
```

Comment:
`ChangeServiceNodePort` is only used from one place, and that caller does not need a static node port. I think we should leave this as-is.